### ISSUE TITLE
across within mutate: Can't select within an unnamed vector

### ISSUE THREAD

**Issue by tungmilan** [tidyverse/dplyr#4867]
Hello,

I tried the new `across()` funtion within `mutate` but got `Can't select within an unnamed vector` error. Please see the reprex below

``` r
library(dplyr, warn.conflicts = FALSE)

DF <- "value
    -0.0970369274475688
    0.131893549586039
    0.0629266495860389
    0.299559132381831
    -0.0128804337656807
    0.0639743960526874
    0.0271669351886552
    0.322395363972391
    0.179591292893632"

DF <- read.table(text = DF, header = TRUE)
```

This worked with `mutate_if`

``` r
DF %>% 
  mutate_if(is.numeric, ~ round(., digits = 1))
#>   value
#> 1  -0.1
#> 2   0.1
#> 3   0.1
#> 4   0.3
#> 5   0.0
#> 6   0.1
#> 7   0.0
#> 8   0.3
#> 9   0.2
```

This gave me error. 

``` r
DF %>% 
  mutate(across(is.numeric, ~ round(., digits = 1)))
#> Error: Can't select within an unnamed vector.

DF %>%
  mutate(across(is.numeric, as.character))
#> Error: Can't select within an unnamed vector.

rlang::last_error()
#> Error: Error: Internal error: Trace data is not square.

rlang::last_error()
# <error/rlang_error>
# Internal error: Trace data is not square.
# Backtrace:
#   1. (function (x, ...) ...
#   2. rlang:::print.rlang_error(x)
#  10. rlang:::format.rlang_error(x, simplify = simplify, fields = fields)
#  12. rlang:::format.rlang_trace(trace, ..., simplify = simplify)
#  13. rlang:::trace_format_branch(x, max_frames, dir, srcrefs)
#  14. rlang:::branch_uncollapse_pipe(trace)
# Run `rlang::last_trace()` to see the full context.

rlang::last_trace()
# <error/rlang_error>
# Internal error: Trace data is not square.
# Backtrace:
#      x
#   1. +-(function (x, ...) ...
#   2. \-rlang:::print.rlang_error(x)
#   3.   +-rlang:::cat_line(format(x, simplify = simplify, fields = fields))
#   4.   | +-base::cat(paste_line(..., .trailing = .trailing), file = file)
#   5.   | \-rlang:::paste_line(..., .trailing = .trailing)
#   6.   |   +-base::paste(chr(...), collapse = "\n")
#   7.   |   \-rlang::chr(...)
#   8.   |     \-rlang::dots_values(...)
#   9.   +-base::format(x, simplify = simplify, fields = fields)
#  10.   \-rlang:::format.rlang_error(x, simplify = simplify, fields = fields)
#  11.     +-base::format(trace, ..., simplify = simplify)
#  12.     \-rlang:::format.rlang_trace(trace, ..., simplify = simplify)
#  13.       \-rlang:::trace_format_branch(x, max_frames, dir, srcrefs)
#  14.         \-rlang:::branch_uncollapse_pipe(trace)

```

What did I miss? Thanks!

**Comment by courtiol** [comment]
It works on a tibble:
``` r
library(dplyr, warn.conflicts = FALSE)

DF <- "value
    -0.0970369274475688
    0.131893549586039
    0.0629266495860389
    0.299559132381831
    -0.0128804337656807
    0.0639743960526874
    0.0271669351886552
    0.322395363972391
    0.179591292893632"

DF <- read.table(text = DF, header = TRUE)

DF %>% 
  tibble() %>%
  mutate(across(is.numeric, ~ round(.x, digits = 1)))
#> # A tibble: 9 x 1
#>   value
#>   <dbl>
#> 1  -0.1
#> 2   0.1
#> 3   0.1
#> 4   0.3
#> 5   0  
#> 6   0.1
#> 7   0  
#> 8   0.3
#> 9   0.2
```

<sup>Created on 2020-02-18 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by courtiol** [comment]
This is caused by a dropping in across():
```
 vars <- tidyselect::eval_select(expr({
        {
            cols
        }
    }), data[, setdiff(names(data), group_vars(data))])
```
Should probably be:
```
 vars <- tidyselect::eval_select(expr({
        {
            cols
        }
    }), data[, setdiff(names(data), group_vars(data)), drop = FALSE])
```
I will test and PR

**Comment by courtiol** [comment]
Problem solved in PR #4867

**Comment by tungmilan** [comment]
Thank you @courtiol for the quick fix!
